addons/account/models/account_move.py

```diff
@@ -2863,10 +2863,11 @@ def _extend_with_attachments(self, attachments, new=False):
             decoder = self._get_edi_decoder(file_data, new=new)
 
             try:
-                if decoder and not success:
-                    with self._get_edi_creation() as invoice:
-                        # pylint: disable=not-callable
-                        success = decoder(invoice, file_data, new)
+                with self.env.cr.savepoint():
+                    if decoder and not success:
+                        with self._get_edi_creation_savepoint() as invoice:
+                            # pylint: disable=not-callable
+                            success = decoder(invoice, file_data, new)
```

Comment:
To me, this part is enough. No need your extra commits

Reply:
I think that part has already been merged and the extra commits are only with crons in localization specific modules. (in case the cron would timeout e.g.)